### PR TITLE
feat(private-database.upgrade): deprecate order/hosting routes

### PR DESCRIPTION
**Pull request by lucas-chaigne**
Signed-off-by: Lucas Chaigne <lucas.chaigne@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | WEB-12807
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

APIv6 routes order/hosting are deprecated. This PR aim is to switch order/hosting/privateSQL route to /order/upgrade/cloudDB for cloudDB and privateSQL upgrade pages.
However, during testing of PR https://github.com/ovh/manager/pull/7899 we found out that some of the order/hosting/privateSQL routes cant be removed (legacy billing privateSQL cannot be upgraded via the new routes).
This PR removes as many calls to deprecated routes as possible

## Translations

- [ ] TRANS-46967

### CODE REVIEW

File: packages/manager/apps/web/client/app/private-database/private-database.service.js [hunk 1]
```diff
@@ -468,94 +468,23 @@ export default class PrivateDatabase {
   /*
         ORDER
     */
-  orderDuration(version, ram) {
```
Comment (marie-j) [inline packages/manager/apps/web/client/app/private-database/private-database.service.js:471]: All those are deleted but I don't see any replacements in controller or whatever class that might call them. Were they not used ?

Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/private-database/private-database.service.js:471]: Hello,

thanks for the review. I think it was used in legacy order private sql pages, but it's deprecated now.
see commit https://github.com/ovh/manager/pull/4865/commits/e0be9d37cf134b0e255d7cb8fd17b86bddc8df15

**Review by antleblanc — DISMISSED** [review]

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 2]
```diff
@@ -58,18 +58,33 @@ angular.module('App').controller(
...
-        remove(
-          this.data.availableRam,
-          (ram) => +ram === +this.database.ram.value,
-        );
-      });
+      this.privateDatabaseService
+        .getUpgradePlans(this.productId)
+        .then((result) => {
+          this.data.availableRam = result.map(
+            (e) => e.planCode.match(/[0-9]+/)?.[0],
```
Comment (marie-j) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:65]: Could we get this information by something less tricky ? Like a product description, technical blob or whatever ?

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 3]
```diff
@@ -58,18 +58,33 @@ angular.module('App').controller(
...
+          );
+          if (this.database.infrastructure === 'legacy') {
+            remove(this.data.availableRam, (ram) => ram === '2048');
```
Comment (marie-j) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:68]: Couldn't this logic be handled by API with available planCode depending on the DB ?

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js (hunk 2 above)
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:65]: thanks for the review. Initially, I wanted to use constants to match plan code and ram, but dynamic extraction was suggested by Yann (see his comment in https://github.com/ovh/manager/pull/7899)

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js (hunk 3 above)
Comment (lucas-chaigne) [reply packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:68]: I think it is already, but DB team (Mikael Davranche) advised me to keep the piece of code to be on the safe side

**Review by anooparveti — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Pull request merged**